### ISSUE TITLE
bug: ReferenceError: Cannot access 'main$1' before initialization

### ISSUE THREAD

**Issue by dnalborczyk** [rollup/rollup#4325]
### Rollup Version

2.62.0

### Operating System (or Browser)

n/a

### Node Version (if applicable)

n/a

### Link To Reproduction

https://rollupjs.org/repl/?version=2.62.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMGZvbyUyMGZyb20lMjAnLiUyRm1haW4uanMnJTNCJTVDbmV4cG9ydCUyMColMjBhcyUyMG5zJTIwZnJvbSUyMCcuJTJGbWFpbi5qcyclM0IlNUNuZXhwb3J0JTIwZGVmYXVsdCUyMCdmb28nJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==

the above link works. this is only a problem when build with `generatedCode: 'es2015'`, which can't [yet] be replicated with the online bundle.

the above is then bundled to:
```
const main = 'foo';

// notice "const" here:
const main$1 = /*#__PURE__*/Object.freeze({
	__proto__: null,
	'default': main,
	ns: main$1
});

export { main as default, main$1 as ns };
```
which throws an error:
```
ReferenceError: Cannot access 'main$1' before initialization
```

**Comment by lukastaegert** [comment]
Definitely a bug, even without es2015 output, as ns would be `undefined` instead of an object. The best solution I could think of here would be to detect these cases and use a getter for `ns`. As a matter of fact, this is already implemented for slightly related cases where you have a circularity across different modules, maybe the check triggering this just needs to be extended: https://rollupjs.org/repl/?version=2.62.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMColMjBhcyUyMG5zJTIwZnJvbSUyMCcuJTJGZm9vLmpzJyUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMmZvby5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJpbXBvcnQlMjAqJTIwYXMlMjBucyUyMGZyb20lMjAnLiUyRm1haW4uanMnJTNCJTVDbmV4cG9ydCUyMGNvbnN0JTIwZm9vJTIwJTNEJTIwJ2ZvbyclM0IlNUNuY29uc29sZS5sb2cobnMpJTNCJTIyJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmVzJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=

**Comment by dnalborczyk** [comment]
> as ns would be undefined instead of an object.

dang, good point! didn't even notice! 😄

**Comment by lukastaegert** [comment]
Fix at #4991
